webapp/src/ts/modules/messages/messages.routes.ts

```diff
@@ -15,6 +15,19 @@ export const routes: Routes = [
         path: '',
         component: MessagesContentComponent
       },
+      {
+        matcher: (url: UrlSegment[]) => {
+          if (url[0]?.path?.indexOf(':') === -1) {
+            return {
+              consumed: url,
+              posParams: {
+                type_id: new UrlSegment('contact:' + url[0].path, {})
```

Comment:
This behaves a little bit differently than the old route when there is more than one param. 
Fe: `http://127.0.0.1/#/messages/uuid/otherparam` effectively dumps the `otherparam` to oblivion, the old route somehow incorporated it.

Reply:
I tried to take a look at the code in master but haven't found it, what is the second param for? I think if we're missing the migration of this, better to create a new ticket to track it 🤔 
Also tried to replicate in gamma but when adding `/otherparam` it redirects to 404, same as in this branch because it's not matching any route. [video](https://user-images.githubusercontent.com/66472237/104988757-77917300-5a4b-11eb-9551-36410aeca599.mp4)

Reply:
Interesting. When I open this link in the browser: https://gamma.dev.medicmobile.org/#/messages/uuid/otherparam it opens this page: 
![image](https://user-images.githubusercontent.com/35681649/104990318-04750600-5a25-11eb-8a61-9c0f67efd245.png)

Reply:
Ah, I can see in your video that you're not opening the same URL I was indicating. 
 https://gamma.dev.medicmobile.org/#/messages/uuid/otherparam vs  https://gamma.dev.medicmobile.org/#/messages/contact:uuid/otherparam

And that's because the redirect handles the 1st url, but the router handles the 2nd. The routers work the same, but the redirect doesn't.

Reply:
Ohh I see it now. That's an interesting url, is it intended or glitch? 
What is this second param useful for?

Reply:
It's not useful, I was just reacting to the fact that your redirect is ignoring any other params that the 1st one, which didn't happen in the old redirect.

Reply:
It looks like the original code that added this redirect was:
```
 $urlRouterProvider.when('/messages/{uuid}', '/messages/contact:{uuid}');
```

So I donno if this just matched _everything_ under the uuid and just appended it in the new URL or just picked the 1st param.

Reply:
Yeah, I think maybe because of the original matching: `uuid:[^:]*` 
I think If in the future we need to support a second param, we can create a ticket and add the support all across from routing to controller according with the needs of the moment. 😊

Reply:
Please add a comment above the redirect, linking to the original issue that added it. Otherwise no-one will know why it's there. 
This is the commit: https://github.com/medic/cht-core/commit/4c6ffce10b61f8dafdbb9d48e6bf9d7d5f36d200 and the issue is: https://github.com/medic/cht-core/issues/4024

Please also add a comment about why we are deliberately ignoring any additional params now vs the old code, which somehow integrated them in the URL.

Reply:
Done added!